src/lib.rs

```diff
@@ -748,7 +748,11 @@ pub enum ConstantInner {
 #[cfg_attr(feature = "arbitrary", derive(Arbitrary))]
 pub enum Binding {
     /// Built-in shader variable.
-    BuiltIn(BuiltIn),
+    BuiltIn {
+        built_in: BuiltIn,
+        /// can only be `true` for [`BuiltIn::Position`]
+        invariant: bool,
```

Comment:
Perhaps there is a better place for this flag to be?
If user locations are to be supported in the future than the current API of this PR wouldn't work, anyway.
If it's only *ever* going to be supported on VS positions, then we could place it on the entry point itself (see `struct EntryPoint`). Would you be interested to check and see how this works out in terms of code complexity?

Reply:
I checked the `EntryPoint` way but I think the code would end up a lot more complex than it is now since the frontends would need to first read the invariant attribute, then look for all entry points that use that position builtin (since it can be in a struct) and set it on the entry point. Backends would also have trouble with this since they would have to map the invariant attribute back.

If the spec changes to allow the attribute to be used on locations, we can change the binding to be a struct with an inner field being the enum we have now and add the invariant flag on the struct.

I agree that it might not be the best spot for it however I tried to put it in multiple places and decided at last to put it here.

Reply:
That's a good insight, thank you!